dotnet/src/SemanticKernel.Core/Extensions/KernelSemanticFunctionExtensions.cs

```diff
@@ -285,4 +313,89 @@ public static class KernelSemanticFunctionExtensions
             kernel.LoggerFactory
         );
     }
+
+    private const string BasedTemplateFactoryAssemblyName = "Microsoft.SemanticKernel.TemplateEngine.Basic";
+    private const string BasedTemplateFactoryTypeName = "BasicPromptTemplateFactory";
+    private static bool s_promptTemplateFactoryInitialized = false;
+    private static Type? s_promptTemplateFactoryType = null;
+
+    /// <summary>
+    /// Create a default prompt template factory.
+    ///
+    /// This is a temporary solution to avoid breaking existing clients.
+    /// There will be a separate task to add support for registering instances of IPromptTemplateEngine and obsoleting the current approach.
+    ///
+    /// </summary>
+    /// <param name="kernel">Kernel instance</param>
+    /// <returns>Instance of <see cref="IPromptTemplateEngine"/>.</returns>
+    private static IPromptTemplateFactory CreateDefaultPromptTemplateFactory(IKernel kernel)
+    {
+#pragma warning disable CS0618 // Type or member is obsolete
+        if (kernel.PromptTemplateEngine is not null)
+        {
+            return new PromptTemplateFactory(kernel.PromptTemplateEngine);
+        }
+#pragma warning restore CS0618 // Type or member is obsolete
+
+        if (!s_promptTemplateFactoryInitialized)
+        {
+            s_promptTemplateFactoryType = GetPromptTemplateFactoryType();
+            s_promptTemplateFactoryInitialized = true;
```

Comment:
You can use just check if the type is null, not sure you need also a boolean flag for it.

Reply:
`GetPromptTemplateFactoryType` could return null which would mean the code would keep trying to get the type. The `s_promptTemplateFactoryInitialized` flag ensures the code only runs once.